### ISSUE TITLE
qpid transport does not close sockets when close() is called

### ISSUE THREAD

**Issue by beav** [celery/kombu#455]
The following code leaves a socket open:

```python
from kombu import Connection

conn = Connection("qpid://localhost:5672/")
conn.connect()
conn.close()
raw_input("Press enter to terminate python process")

```
output of `netstat -anp | grep <python pid>` while process is running:
```
tcp        0      0 127.0.0.1:46353         127.0.0.1:5672          ESTABLISHED 32373/python
```

Eventually, opening and closing the connection will result in the system running out of file handles.

**Comment by bmbouter** [comment]
PR available at  #456

**Comment by bmbouter** [comment]
This has been resolved.

**Comment by bmbouter** [comment]
There was an additional regression that needed fixing with this issue. It has been fixed with PR #485
